src/base/header.rs

```diff
@@ -407,15 +407,15 @@ impl FromStr for Flags {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         let mut flags = Flags::new();
-        for token in s.to_uppercase().split(' ') {
+        for token in s.split(' ') {
```

Comment:
Not exactly the prettiest thing, but to_uppercase() does allocate, and we only need to compare some basic ASCII characters.
We could probably be smarter about it, and just uppercase the ascii range, which is doable, but this seems to just be simple enough to not have to worry about it.